src/DNA.cpp

```diff
@@ -489,3 +549,108 @@ string DNA::get_binary_representation(int value, int bit_count) {
 
     return output;
 }//end get_binary_representation
+
+/**
+ * rand_num
+ * 
+ * Returns a random integer [start, end).
+ **/
+int DNA::rand_num(int start, int end) {
+    int diff = end - start;
+    int r = rand() % diff;
+    r += start;
+    return r;
+}//end rand_num
+
+/**
+ * clip
+ * 
+ * Returns only the bits of value between [start, end). 
+ * Assumed that start <= end <= 64;
+ **/
+dna_t DNA::clip(dna_t value, int start, int end) {
+    long s = 0xffffffffffffffff << start;
+    long e;
+    if (end == 64 ) {
+        e = 0xffffffffffffffff;
+    } else {
+        e = 0x7fffffffffffffff >> (63 - end);
+    }
+    return (s & e) & value;
+}//end clip
+
+/**
+ * multipoint_crossover
+ * 
+ * Performs a multipoint crossover of the two dna 
```

Comment:
Reader might like a little refresh on what a multipoint crossover is.

Reply:
Fixed in next commit.